internal/controller/provisioner/objects_test.go

```diff
@@ -1058,3 +1068,106 @@ func TestBuildNginxConfigMaps_WorkerConnections(t *testing.T) {
 	g.Expect(ok).To(BeTrue())
 	g.Expect(bootstrapCM.Data["main.conf"]).To(ContainSubstring("worker_connections 2048;"))
 }
+
+func TestBuildReadinessProbe(t *testing.T) {
+	t.Parallel()
+	g := NewWithT(t)
+
+	defaultProbe := &corev1.Probe{
+		ProbeHandler: corev1.ProbeHandler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Path: "/readyz",
+				Port: intstr.FromInt32(dataplane.DefaultNginxReadinessProbePort),
+			},
+		},
+		InitialDelaySeconds: 3,
+	}
+
+	provisioner := &NginxProvisioner{}
+
+	tests := []struct {
+		nProxyCfg *graph.EffectiveNginxProxy
+		expected  *corev1.Probe
+		name      string
+	}{
+		{
+			name:      "nginx proxy config is nil, default probe is returned",
+			nProxyCfg: nil,
+			expected:  defaultProbe,
+		},
+		{
+			name: "deployment is nil, default probe is returned",
+			nProxyCfg: &graph.EffectiveNginxProxy{
+				Kubernetes: &ngfAPIv1alpha2.KubernetesSpec{
+					Deployment: nil,
+				},
+			},
+			expected: defaultProbe,
+		},
+		{
+			name: "daemonSet is nil, default probe is returned",
+			nProxyCfg: &graph.EffectiveNginxProxy{
+				Kubernetes: &ngfAPIv1alpha2.KubernetesSpec{
+					DaemonSet: nil,
+				},
+			},
+			expected: defaultProbe,
```

Comment:
These are technically the same configurations. Could probably just combine them.